stake-pool/program/src/instruction.rs

```diff
@@ -92,18 +94,79 @@ pub enum StakePoolInstruction {
     ///  10. `[]` Stake program id,
     RemoveValidatorFromPool,
 
-    ///   Updates balances of validator stake accounts in the pool
+    /// (Staker only) Split a validator stake account into a transient stake account.
     ///
-    ///   0. `[w]` Validator stake list storage account
-    ///   1. `[]` Sysvar clock account
-    ///   2. ..2+N ` [] N validator stake accounts to update balances
+    /// In order to rebalance the pool without taking custody, the staker needs
+    /// a way of reducing the stake on a stake account. This instruction splits
+    /// some amount of stake, up to the total activated stake, from the canonical
+    /// validator stake account, into its "transient" stake account, defined by:
+    ///
+    /// ```
+    /// Pubkey::find_program_address(
+    ///     &[&stake_account_address.to_bytes()[..32],], program_id,
+    /// )
+    /// ```
+    ///
+    /// After the split, the instruction automatically deactivates the stake.
+    ///
+    /// The instruction only succeeds if the transient stake account does not
+    /// exist. The amount of lamports to move must be at least rent-exemption
+    /// plus 1 lamport.
+    ///
+    ///  0. `[]` Stake pool
+    ///  1. `[s]` Stake pool staker
+    ///  2. `[]` Validator list
+    ///  3. `[]` Stake pool withdraw authority
+    ///  5. `[w]` Canonical stake account to split from
+    ///  5. `[w]` Transient stake account to receive split
+    ///  6. `[]` Clock sysvar
+    ///  7. `[]` Rent sysvar
+    ///  8. `[]` System program
+    ///  9. `[]` Stake program
+    ///  userdata: amount of lamports to split
+    ValidatorToReserve(u64),
```

Comment:
I don't love the instruction names `ValidatorToReserve` and `ReserveToValidator`, I was just being lazy in my initial writeup.

Maybe `IncreaseValidatorStake` and `DecreaseValidatorStake`.   Any other ideas?

Reply:
D'oh, I thought you wanted those names.  I like `IncreaseValidatorStake` and `DecreaseValidatorStake`, since they're clear opposites that accurately describe what will happen in the future.  The only other options I could come up with that describe what's happening in the instruction are `DelegateReserveStakeToValidator` and `DeactivateValidatorStakeToReserve`

Reply:
heh, na.  I came up with variations of `DelegateReserveStakeToValidator` as well and it seemed like a mouthful!